src/server/cluster/cluster_slot_migration.cc

```diff
@@ -82,4 +100,73 @@ ClusterSlotMigration::Info ClusterSlotMigration::GetInfo() const {
   return {ctx.host, ctx.port, state_};
 }
 
+void ClusterSlotMigration::MainMigrationFb() {
+  VLOG(1) << "Main migration fiber started";
+
+  state_ = ClusterSlotMigration::C_FULL_SYNC;
+
+  // TODO add reconnection code
+  if (auto ec = InitiateDflyFullSync(); ec) {
+    LOG(WARNING) << "Error syncing with " << server().Description() << " " << ec << " "
+                 << ec.message();
+  }
+}
+
+std::error_code ClusterSlotMigration::InitiateDflyFullSync() {
```

Comment:
I believe that in cluster slot migration we do not need to wait for flows to finish their full sync to move to stable sync. This part is probably missing in the design doc. But from how I see it we can send from each flow the request to sync on the slots. Once the shard finished with sending all the data it should send that it finished its fullsync. Once all flows finished full sync the source node can move to slot migration state stable sync. Note that we do not need an rdb loader in the slot migration as the migraion is not done in rdb format but with restore commads so we will only need the executer that we have in the replication code. We will need to add an opcode for in the jounal change that will notify that this is the last store command send to show that full sync in shard finished.

Reply:
I can't add anything here, because it depends on our requirements or design. I haven't read it yet and I don't really understand how all threads can be in stable state for a migration? Doesn't it mean we finished it?

Replication has full sync only to track transition "completeness", meaning there is a phase we reach where we lift the `LOADING` state and make the instance respond to requests

Reply:
When all migration flows are in stable this does not mean we finished the migration. We need to wait for the cluster config command to finish the migration so that all nodes will be aware of the new slot ownership. It is a bit similar to takeove command that once takeover command is executed we can finish the replication of the data and the replica is the new master.

Reply:
I will do change state to stable sync in the next PR